cppcache/include/geode/Region.hpp

```diff
@@ -114,7 +114,7 @@ class _GEODE_EXPORT Region : public std::enable_shared_from_this<Region> {
 
   /** Return the RegionAttributes for this region.
    */
-  virtual std::shared_ptr<RegionAttributes> getAttributes() const = 0;
+  virtual RegionAttributes getAttributes() const = 0;
```

Comment:
`RegionAttributes` are immutable so it might make a lot more sense to return a `const RegionAttributes&` here. It would fix the issue you addressed in a test change where you were taking the ref to the region name off an r-value RegionAttributes.